pfcp/handler/handler.go

```diff
@@ -354,14 +354,12 @@ func HandlePfcpAssociationReleaseRequest(msg *udp.Message) {
 	nodeID := smf_context.NewNodeID(nodeIDStr)
 
 	upf := smf_context.RetrieveUPFNodeByNodeID(*nodeID)
-	var cause uint8
-	if upf != nil {
-		smf_context.RemoveUPFNodeByNodeID(*nodeID)
-		cause = ie.CauseRequestAccepted
-	} else {
-		cause = ie.CauseNoEstablishedPFCPAssociation
+	if upf == nil {
+		logger.PfcpLog.Errorf("can't find UPF[%s]", nodeIDStr)
+		return
 	}
-
+	smf_context.RemoveUPFNodeByNodeID(*nodeID)
+	cause := ie.CauseRequestAccepted
```

Comment:
Can you please explain this change? Is it because we can only receive `association release request` if there was already an `association setup request`? If so, could there be a case where we (smf) receive an `association release request` without having an `association setup` already in place?

Reply:
if upf is nil, we should not even try to call `SendPfcpAssociationReleaseResponse` because this call is made using `upf.Port` (which would have caused a nil pointer). This is why we not return if upf is nil.